src/IssueViz.Security/IssuesStore/IssuesStoreToIssueLocationStoreAdapter.cs

```diff
@@ -0,0 +1,126 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using SonarLint.VisualStudio.Core.Helpers;
+using SonarLint.VisualStudio.IssueVisualization.Editor.LocationTagging;
+using SonarLint.VisualStudio.IssueVisualization.Models;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Security.IssuesStore
+{
+    /// <summary>
+    /// Adapter between <see cref="IIssuesStore"/> and <see cref="IIssueLocationStore"/>
+    /// </summary>
+    internal interface IIssuesStoreToIssueLocationStoreAdapter : IIssueLocationStore, IDisposable
+    {
+    }
+
+    /// <summary>
+    /// MEF-imports <see cref="IIssuesStore"/> and implements <see cref="IIssueLocationStore"/> on top of the given stores.
+    /// </summary>
+    [Export(typeof(IIssuesStoreToIssueLocationStoreAdapter))]
+    [Export(typeof(IIssueLocationStore))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal sealed class IssuesStoreToIssueLocationStoreAdapter : IIssuesStoreToIssueLocationStoreAdapter
+    {
+        private readonly IEnumerable<IIssuesStore> issueStores;
+
+        [ImportingConstructor]
+        public IssuesStoreToIssueLocationStoreAdapter([ImportMany] IEnumerable<IIssuesStore> issueStores)
+        {
+            this.issueStores = issueStores;
+
+            foreach (var store in issueStores)
+            {
+                store.IssuesChanged += Store_IssuesChanged;
+            }
+        }
+
+        public event EventHandler<Editor.LocationTagging.IssuesChangedEventArgs> IssuesChanged;
+
+        public IEnumerable<IAnalysisIssueLocationVisualization> GetLocations(string filePath)
+        {
+            var matchingLocations = issueStores
+                .SelectMany(store => store.GetAll())
+                .SelectMany(issueViz => issueViz.GetAllLocations())
+                .Where(locationViz => !string.IsNullOrEmpty(locationViz.CurrentFilePath) &&
+                                      PathHelper.IsMatchingPath(locationViz.CurrentFilePath, filePath));
+
+            return matchingLocations;
+        }
+
+        public void Refresh(IEnumerable<string> affectedFilePaths)
+        {
+            // Implementation is not required:
+            // This method was originally added in order to notify the error list sinks that the taggers changed the span of the issue visualizations.
+            // The newer issue stores display their data in plain xaml lists, and the span changes are tracked via NotifyPropertyChanged mechanism in IAnalysisIssueVisualization.
+        }
+
+        public bool Contains(IAnalysisIssueVisualization issueVisualization)
```

Comment:
Thinking about it, `Contains` was added to fix the selection bug in the Error List, wasn't it? If the problem only exists for the Error List, does the `Contains` method actually need to be on the `IIssueLocation` interface?

Assuming that is the case, we should fix it separately (although it would mean there might be no point in implementing or testing the method here).

Reply:
Hmm, the problem was that the selection service (and the panel) displayed an issue that was no longer there. 
We haven't added secondary locations yet to taint and hotspots, so it's difficult to know if the same problem would exist there or not, but it's a good point.
I can think of 2 similar scenarios:
1. Select a taint issue that has secondary locations and then re-sync the binding so that the issue no longer exists.
2. Select a hotspot that has secondary locations and then remove it from the list.

Depending on the implementation of these two, there might be a selection bug and might not.
As you may recall, the original version of the error list store cleared the selection each time a new analysis was performed. This is similar to the 2 scenarios above; if there's no "monitor" service, then we would need to clear the selection. It doesn't sound so bad honestly... I hope that the monitor and the contains method can both go away when we change the selection.